contracts/smart-wallet/utils/BaseAccountFactory.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+// Utils
+import "../../extension/Multicall.sol";
+import "../../openzeppelin-presets/proxy/Clones.sol";
+import "../../openzeppelin-presets/utils/structs/EnumerableSet.sol";
+import "../utils/BaseAccount.sol";
+
+// Interface
+import "../interfaces/IEntrypoint.sol";
+import "../interfaces/IAccountFactory.sol";
+
+//   $$\     $$\       $$\                 $$\                         $$\
+//   $$ |    $$ |      \__|                $$ |                        $$ |
+// $$$$$$\   $$$$$$$\  $$\  $$$$$$\   $$$$$$$ |$$\  $$\  $$\  $$$$$$\  $$$$$$$\
+// \_$$  _|  $$  __$$\ $$ |$$  __$$\ $$  __$$ |$$ | $$ | $$ |$$  __$$\ $$  __$$\
+//   $$ |    $$ |  $$ |$$ |$$ |  \__|$$ /  $$ |$$ | $$ | $$ |$$$$$$$$ |$$ |  $$ |
+//   $$ |$$\ $$ |  $$ |$$ |$$ |      $$ |  $$ |$$ | $$ | $$ |$$   ____|$$ |  $$ |
+//   \$$$$  |$$ |  $$ |$$ |$$ |      \$$$$$$$ |\$$$$$\$$$$  |\$$$$$$$\ $$$$$$$  |
+//    \____/ \__|  \__|\__|\__|       \_______| \_____\____/  \_______|\_______/
+
+abstract contract BaseAccountFactory is IAccountFactory, Multicall {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /*///////////////////////////////////////////////////////////////
+                                State
+    //////////////////////////////////////////////////////////////*/
+
+    BaseAccount internal immutable _accountImplementation;
+
+    mapping(address => EnumerableSet.AddressSet) internal accountsOfSigner;
+    mapping(address => EnumerableSet.AddressSet) internal signersOfAccount;
+
+    /*///////////////////////////////////////////////////////////////
+                            Constructor
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(BaseAccount accountImpl) {
```

Comment:
do we actually use any `BaseAccount` functions here? Feels like that can just be of type `address`, would make the API much simpler and doesn't restrict what you can pass here